DBSCE/Overview.md

```diff
@@ -0,0 +1,275 @@
+
+### DBSC(E) use cases
+
+This section expands on the [generic design](#high-level-design) to address different enterprise use cases:
+
+#### IDP is RP and Calls Public Local Key Helper
+
+This is the same use case elaborated in the high level design [above](#high-level-design). However, we have separated the IdP/RP components of the server that applies to certain enterprise and consumer use cases in the below diagram.
+
+![IDPSameAsRP-CallsPublicLocalKeyHelper](./IDPSameAsRP-CallsPublicLocalKeyHelper.svg)
+
+Highlights:
+
+- Auth tokens are not mentioned in the DBSC(E) high level design to simplify the over all scenario. However, most signin/access operations will often make use of auth tokens and issue cookies based on those tokens. In such case, the IdP can generate the auth tokens and embed the thumbprint of the public key in the token.
+- The tokens can be delivered to the RP directly through an API instead of a header based response.
+- The tokens also contain the thumbbrint of the public key, and the RP must validate the thumbprint from the token against the thumbprint from the JWT. The RP will also embed the thumbprint in the cookie for subsequent session validation.
+
+#### IDP Calls Public Local Key Helper
+
+In many use cases, it is also a valid to have separate servers as [RP](#relying-party-rp) and [IdP](#identity-provider-idp). We address the use case where the IdP calls a _public Local Key Helper_ and is a separate service from the RP below.
+
+For easy mapping with the existing DBSC proposal, please note:
+
+- Steps 1-16 specify the key generation process for a public local key helper.
+- Steps 17-29 are [DBSC](https://github.com/wicg/dbsc), added for completeness.
+
+
+![IDPCallsPublicLocalKeyHelper](./IDPCallsPublicLocalKeyHelper.svg)
+
+#### IDP Calls Private Local Key Helper
+
+As referred in [Local Key Helper](#local-key-helper) section, it is possible for an IdP to implement a private local key helper and establish a private protocol to communicate. Any such private local key helper can only be used by the specific IDP that owns its implementation.
+
+In DBSC(E), the use of private local key helper for specific IDPs enables the below optimizations:
+
+- Since the IDP can trust the private local key helper, the `HelperId` discovery can be optimized by caching. An IDP can specify its preferred ordered list of `HelperId`s in any response, and can request the browser to cache its preference for a specific duration(`HelperCacheTime`). Also, such requests need not be initiated by the RP that implements DBSC(E), the IDP can specify its `HelperIdList` in any communication(request/response) within the browser context.
```

Comment:
Maybe just a small clarification:

> An IDP can specify its preferred ordered list of `HelperId`s in any response, **even if it's not DBSC(E) enabled**, and can request the browser to cache its preference for a specific duration(`HelperCacheTime`)

Another thing is that HelperCacheTime can be used in other places too, not only for private local key helpers, we just cannot do it ahead of time like in private case, because we need to generate nonce anyway before we redirect to the attestation service, so we're not saving a redirect by caching. It won't necessarily be always same helper as well depending on the username that user enters.

Reply:
Done.